src/openms/source/ANALYSIS/SVM/SimpleSVM.cpp

```diff
@@ -107,55 +111,80 @@ void SimpleSVM::setup(PredictorMap& predictors, const map<Size, Int>& labels)
         " observations.";
       throw Exception::InvalidValue(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
                                     msg, String(it->first));
     }
-    data_.x[index] = &(nodes_[it->first][0]);
-    data_.y[index] = double(it->second);
-    label_table[it->second]++;
+    data_.x[index] = &(nodes_[training_index][0]);
+    data_.y[index] = label;
+    label_table[label]++;
   }
-  if (label_table.size() < 2)
+
+  if (classification)
   {
-    throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
-                                        "Need at least two classes (distinct "
-                                        "labels) for SVM classification.");
+    // check for 2 or more classes
+    if (label_table.size() < 2)
+    {
+      throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
+                                          "Need at least two classes (distinct "
+                                          "labels) for SVM classification.");
+    }
+
+    String msg = "Training SVM on " + String(data_.l) + " observations. Classes:";
+    for (map<double, Size>::iterator it = label_table.begin(); 
+        it != label_table.end(); ++it)
+    {
+      if (it->second < n_parts_)
+      {
+        msg = "Not enough observations of class " + String(it->first) + " for " +
+          String(n_parts_) + "-fold cross-validation.";
+        throw Exception::MissingInformation(__FILE__, __LINE__, 
+                                            OPENMS_PRETTY_FUNCTION, msg);
+      }
+      msg += "\n- '" + String(it->first) + "': " + String(it->second) +
+        " observations";
+    }
+    OPENMS_LOG_INFO << msg << endl;
+
+    svm_params_.svm_type = C_SVC;
   }
-  String msg = "Training SVM on " + String(data_.l) + " observations. Classes:";
-  for (map<Int, Size>::iterator it = label_table.begin(); 
-       it != label_table.end(); ++it)
-  {
-    if (it->second < n_parts_)
+  else
+  { // regression
+    if (data_.l < n_parts_) // TODO: check minimum amount of points needed for training a regression model. Assume 1 is enough for now.
     {
-      msg = "Not enough observations of class " + String(it->first) + " for " +
-        String(n_parts_) + "-fold cross-validation.";
-      throw Exception::MissingInformation(__FILE__, __LINE__, 
-                                          OPENMS_PRETTY_FUNCTION, msg);
+        String msg = "Not enough observations for " + String(n_parts_) + "-fold cross-validation.";
+        throw Exception::MissingInformation(__FILE__, __LINE__, 
+                                            OPENMS_PRETTY_FUNCTION, msg);
     }
-    msg += "\n- '" + String(it->first) + "': " + String(it->second) +
-      " observations";
+
+    OPENMS_LOG_INFO << "Training SVR on " + String(data_.l) + " observations." << endl;
+
+    svm_params_.svm_type = EPSILON_SVR;
+    svm_params_.p = 0.1; // epsilon parameter of epsilon-SVR
```

Comment:
Doesn't the epsilon parameter need tuning as well (together with C)?
Shouldn't be hard to do because the RBF SVM optimises over two parameters, too.